sphinxcontrib/datatemplates/directive.py

```diff
@@ -171,6 +179,13 @@ def run(self):
             template = '\n'.join(self.content)
             render_function = _templates(builder).render_string
 
+        if not template:
+            error = self.state_machine.reporter.error(
+                "Template is empty",
+                nodes.literal_block(self.block_text, self.block_text),
+                line=self.lineno)
+            return [error]
```

Comment:
What does return an error do that's different from raising the exception?

Reply:
If I used multiple different directives in my docs, it wouldn't be clear to me which one caused the exception. By returning an error in the `run()` method, Sphinx would log it with the path to the file and a line number at which it occurred. It is the right way to handle errors in Sphinx.

Reply:
Nice, that definitely sounds like an improvement!